lib/src/main/java/com/dynatrace/file/util/PollBasedFilePoller.java

```diff
@@ -59,21 +74,86 @@ public boolean fileContentsUpdated() {
     return changedSinceLastInquiry.getAndSet(false);
   }
 
+  private synchronized void initialPoll() {
+    try {
+      prevChecksum = getFileChecksum();
+      prevModMillis =
+          Files.getLastModifiedTime(absoluteFilePath, LinkOption.NOFOLLOW_LINKS).toMillis();
+    } catch (IOException e) {
+      logger.warning(
+          () ->
+              String.format(
+                  "Failed to read file %s, stopping polling. Error: %s", absoluteFilePath, e));
+      worker.cancel(true);
+    }
+  }
+
   private synchronized void poll() {
     try {
-      FileTime lastModifiedAt = Files.getLastModifiedTime(absoluteFilePath);
+      // check if the filetime has changed.
+      long lastModifiedMillis = Files.getLastModifiedTime(absoluteFilePath).toMillis();
+      if (lastModifiedMillis > prevModMillis) {
+        prevModMillis = lastModifiedMillis;
+        changedSinceLastInquiry.set(true);
+        return;
+      }
 
-      if (lastModifiedAt.compareTo(prevModifiedAt) >= 0) {
-        if (prevModifiedAt.compareTo(ZERO_FILE_TIME) != 0) {
-          changedSinceLastInquiry.set(true);
-        }
+      // if the filetime hasn't changed, check if the hash has changed.
```

Comment:
```suggestion
      // If the filetime hasn't changed, check if the hash has changed.
      // Certain platform implementations don't reflect the modified date properly.
```
(If I understood the issue correctly)